From c4ed0484b3c2091eb881f7c93d10321c0fde5690 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 10 Feb 2021 19:56:15 +0200
Subject: [PATCH] [SW-31383] gaudi: security: fix dbi lock

The DBI lock function was written and worked correctly by its time.
Later we have discovered several security holes and those were addressed
in a certain way, but the preboot_security_dbi_lock() function was left
unaddressed and thus after the closure of security holes started working
incorrectly. Specifically, the RR was not updated per the new RRs
settings and the preboot_secure_lbw_rr() function was changed and
propagated the incorrect configuration to more domains (e.g. DMA_IF).
Fix the preboot_security_dbi_lock() function to be alinged with the RRs
configuration and new preboot_secure_lbw_rr() function.

Change-Id: I3feaa00722b0a91ab56e11545ed638dc52b066b9
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 1735946..a5c7c24 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1207,6 +1207,11 @@
 	max = mmPCIE_WRAP_BASE;
 	preboot_secure_lbw_rr(7, min, max);
 
+	/*
+	 * Note: we skip here the PCIE_DBI, since iATU might not be configured
+	 * yet and thus host needs access to DBI/iATU space.
+	 * The DBI is closed at a later stage in DBI lock function.
+	 */
 	/* PCIE_CORE up to PCIE_CORE_L: RR_8 */
 	min = mmPCIE_CORE_BASE - 1;
 	max = HL_CFG_BASE + mmPCIE_CORE_MSI_REQ;
@@ -1615,9 +1620,8 @@
 	if (!preboot_security_is_enabled())
 		return;
 
-	/* PCIE_DBI up to MMU_UP: RR_4 */
+	/* PCIE_DBI up to PCIE_CORE_L: RR_8 */
 	min = mmPCIE_DBI_BASE - 1;
-	max = mmMMU_UP_BASE;
-	preboot_secure_lbw_rr(4, min, max);
-	preboot_secure_pcie_rr(4, min, max);
+	max = HL_CFG_BASE + mmPCIE_CORE_MSI_REQ;
+	preboot_secure_lbw_rr(8, min, max);
 }
